### PR TITLE
RSP-1701: Clean up for Tooltips

### PR DESCRIPTION
**Pull request by pst67662**
Closes https://jira.corp.adobe.com/browse/RSP-1701

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@adobe/spectrum-css-temp/components/tooltip/index.css [hunk 1]
```diff
@@ -23,8 +23,6 @@ governing permissions and limitations under the License.
   @inherit: %spectrum-overlay;
 
   position: relative;
-  left: 0px;
-  top: 0px;
```
Comment (pst67662) [inline packages/@adobe/spectrum-css-temp/components/tooltip/index.css:27]: removed: left and top placement tooltips are stretched

File: packages/@react-spectrum/tooltip/stories/Tooltip.stories.tsx [hunk 2]
```diff
@@ -53,32 +53,23 @@ storiesOf('Tooltip', module)
     () => render(longMarkup)
   )
   .add(
-    'triggered by click, placement: left',
-    () => renderWithTrigger('This is a tooltip.', {placement: 'start', type: 'click'})
+    'triggered, placement: left',
+    () => renderWithTrigger('This is a tooltip.', {placement: 'start'})
```
Comment (pst67662) [inline packages/@react-spectrum/tooltip/stories/Tooltip.stories.tsx:57]: do we want separate stories for Tooltip and TooltipTrgger?

**Comment by pst67662** [comment]
bottom/top tooltip slides diagonally when open for the first time... need to check it

**Comment by snowystinger** [comment]
From testing:
- can't mouse over the tooltip and have it stay up (don't know if that's a requirement)
- if i click on the trigger, it'll cause the tooltip to go away and put it into a state where the animations and timers don't seem quite right anymore
- disabled trigger story doesn't seem to have a disabled button?

File: packages/@react-aria/tooltip/src/useTooltip.ts [hunk 3]
```diff
@@ -10,45 +10,24 @@
...
-    if (contextProps.onPointerLeave && contextProps.onPointerEnter) {
-      tooltipProps.onPointerLeave = contextProps.onPointerLeave;
-      tooltipProps.onPointerEnter = contextProps.onPointerEnter;
-    }
-    if (contextProps.onMouseLeave && contextProps.onMouseEnter) {
-      tooltipProps.onMouseLeave = contextProps.onMouseLeave;
-      tooltipProps.onMouseEnter = contextProps.onMouseEnter;
-    }
-  }
-
```
Comment (LFDanLu) [inline packages/@react-aria/tooltip/src/useTooltip.ts:50]: If I remember correctly, this code here was to provide the tooltip with the same mouseEnter and mouseLeave handlers that the trigger button had for displaying and hiding the tooltip upon hovering the trigger button. This made it so if the user hovered over the tooltip while it was displayed, the tooltip would remain visible instead of hiding.

Might be possible to use the new tooltiptrigger context to send the trigger props to the tooltip now? Need to investigate

File: packages/@react-spectrum/tooltip/stories/Tooltip.stories.tsx (hunk 2 above)
Comment (MidnightCoder06) [reply packages/@react-spectrum/tooltip/stories/Tooltip.stories.tsx:57]: Yes I think that would a good idea. Following menu / menu trigger & dialog / dialog trigger pattern.

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js [hunk 4]
```diff
@@ -34,81 +33,6 @@ describe('TooltipTrigger', function () {
     cleanup();
   });
 
-  describe('click related tests', function () {
```
Comment (MidnightCoder06) [inline packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js:37]: Why did you remove these tests?

Comment (MidnightCoder06) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js:37]: Adding some tests for focus behavior is likely the only change that needs to be made to the tests.

File: packages/@react-spectrum/tooltip/stories/Tooltip.stories.tsx [hunk 5]
```diff
@@ -53,32 +53,23 @@ storiesOf('Tooltip', module)
...
-     'single tooltip: hover',
-     () => renderMultipleTriggers('This is a tooltip.', {placement: 'left', type: 'hover'})
-   ).add(
-     'single tooltip: click',
-     () => renderMultipleTriggers('This is a tooltip.', {placement: 'left', type: 'click'})
-   ).add(
-     'single tooltip: hover & click',
-     () => renderWithDifferentTriggerTypes('This is a tooltip.')
+     'multiple tooltips',
+     () => renderMultipleTriggers('This is a tooltip.', {placement: 'left'})
```
Comment (MidnightCoder06) [inline packages/@react-spectrum/tooltip/stories/Tooltip.stories.tsx:72]: what kind of triggers? These seem like hover triggers but clicking is causing odd behavior.

File: packages/@react-aria/tooltip/src/useTooltip.ts (hunk 3 above)
Comment (LFDanLu) [reply packages/@react-aria/tooltip/src/useTooltip.ts:50]: Yep, did some digging and you can resolve the tooltip hovering issue by doing the following:
- spread `...hoverProps` in `tooltipProps` in `useTooltipTrigger`
- propagate those hoverProps to the tooltip by doing `let {tooltipProps} = useTooltip(otherProps)` in `Tooltip.tsx` (prob could refine what exactly to pass to useTooltipProps)
- grab `otherProps` from `useTooltip` and extract `onPointerLeave/onPointerEnter` from it. Add those to the tooltipProps that useTooltip returns

I see that the ticket says that we want to have `useTooltip` handle role/id only so maybe the logic to get `onPointerLeave/onPointerEnter` should be done in Tooltip instead, dunno

File: packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js (hunk 4 above)
Comment (dannify) [reply packages/@react-spectrum/tooltip/test/TooltipTrigger.test.js:37]: We are not supporting click functionality for Tooltips. They should only open on hover and focus interactions so we won't need these tests.

File: packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx [hunk 6]
```diff
@@ -10,78 +10,73 @@
...
+      </DOMPropsResponder>
+      <Overlay isOpen={state.open}>
+        <TooltipContext.Provider
+          value={{
+            placement,
+            overlayRef,
+            UNSAFE_style: overlayProps.style,
+            isOpen: state.open,
+            ...tooltipProps
+          }}>
```
Comment (LFDanLu) [inline packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx:76]: Hrm, digging into this currently but it seems like the diagonal tooltip sliding that happens on first render is due to the change from using cloneElement to using this context, maybe something with how the refs?

Comment (LFDanLu) [reply packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx:76]: After digging into this some more, it seems the diagonal sliding is happening because the tooltip is rendered out of position initially and `isOpen` is true while it is still trying to get the tooltip to the right place. I got the sliding to go away by appending doing a pretty silly conditional `'is-open': isOpen && styleProps.style.maxHeight` in `Tooltip.tsx` but I haven't dug into why isOpen is false initially for the original code

Comment (LFDanLu) [reply packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx:76]: Ah, its because of how `Overlay`-> `OpenTransition`->`Transition` provides a `isOpen` prop to its children via cloneElement. To fix I think you'll have to change the ordering here so the `TooltipContext.Provider` wraps `Overlay` instead, then you'll have drop the `isOpen` you pass via the TooltipProvider here so it doesn't overwrite the `props.isOpen` in Tooltip provided by the cloneElement that happens in `OpenTransition`.

```
        <TooltipContext.Provider
          value={{
            placement,
            overlayRef,
            UNSAFE_style: overlayProps.style,
            ...tooltipProps
          }}>
            <Overlay isOpen={state.open}>
              {tooltip}
            </Overlay>
        </TooltipContext.Provider>
```

```

```

Comment (pst67662) [reply packages/@react-spectrum/tooltip/src/TooltipTrigger.tsx]: thnaks! that was very helpful :) reordered provider without isOpen and it works well https://github.com/adobe-private/react-spectrum-v3/pull/426/commits/22f7fff1a6ef53e94c7c9b051c62ee2ed5733e6e

**Comment by pst67662** [comment]
> From testing:
> 
> * can't mouse over the tooltip and have it stay up (don't know if that's a requirement)
> * if i click on the trigger, it'll cause the tooltip to go away and put it into a state where the animations and timers don't seem quite right anymore
> * disabled trigger story doesn't seem to have a disabled button?

@snowystinger 

- mouse over tooltip fixed
- timers and animations have followup task https://jira.corp.adobe.com/browse/RSP-1367
- I think that isDisabled story only disables triggering logic, will not affect button (overaly trigger also does that)... any suggestion?

File: packages/@react-aria/tooltip/src/useTooltip.ts [hunk 7]
```diff
@@ -10,45 +10,29 @@
...
-    if (contextProps.onPointerLeave && contextProps.onPointerEnter) {
-      tooltipProps.onPointerLeave = contextProps.onPointerLeave;
-      tooltipProps.onPointerEnter = contextProps.onPointerEnter;
-    }
-    if (contextProps.onMouseLeave && contextProps.onMouseEnter) {
-      tooltipProps.onMouseLeave = contextProps.onMouseLeave;
-      tooltipProps.onMouseEnter = contextProps.onMouseEnter;
-    }
-  }
+  let {onPointerLeave, onPointerEnter} = useContext(DOMPropsResponderContext) || {};
```
Comment (devongovett) [inline packages/@react-aria/tooltip/src/useTooltip.ts:28]: I would make sure to pass all props through here, not just these two. There will be more in other cases, like `onMouseLeave`/`onMouseEnter`, etc. Best to just pass them all through.
